app/helpers/gtl_helper.rb

```diff
@@ -0,0 +1,125 @@
+module GtlHelper
+  def gtl_selected_records
+    records = @edit[:object_ids] unless @edit.nil? || @edit[:object_ids].nil?
+    records = @edit[:pol_items] unless @edit.nil? || @edit[:pol_items].nil?
+    records = params[:rec_ids] unless params.nil? || params[:rec_ids].nil?
+    records = records.map(&:to_i) if !records.nil? && records.first.kind_of?(String)
+    records = records.map(&:id) if !records.nil? && !records.first.kind_of?(Integer)
+    records
+  end
+
+  def miq_tile_view
+    content_tag(
+      'miq-tile-view', '',
+      "ng-if"            => "dataCtrl.gtlType === 'grid' || dataCtrl.gtlType === 'tile'",
+      "ng-class"         => "{'no-action': dataCtrl.initObject.showUrl === ''}",
+      "settings"         => "dataCtrl.settings",
+      "per-page"         => "dataCtrl.perPage",
+      "rows"             => "dataCtrl.gtlData.rows",
+      "on-row-click"     => "dataCtrl.onItemClicked(item, event)",
+      "on-sort"          => "dataCtrl.onSort(headerId, isAscending)",
+      "on-item-selected" => "dataCtrl.onItemSelect(item, isSelected)",
+      "load-more-items"  => "dataCtrl.onLoadNext(start, perPage)",
+      "columns"          => "dataCtrl.gtlData.cols",
+      "type"             => "dataCtrl.gtlType === 'grid' ? 'small' : 'big'"
+    )
+  end
+
+  def miq_data_table
+    content_tag(
+      'miq-data-table', '',
+      "ng-if"            => "dataCtrl.gtlType === 'list'",
+      "ng-class"         => "{'no-action': dataCtrl.initObject.showUrl === false}",
+      "settings"         => "dataCtrl.settings",
+      "per-page"         => "dataCtrl.perPage",
+      "rows"             => "dataCtrl.gtlData.rows",
+      "on-row-click"     => "dataCtrl.onItemClicked(item, event)",
+      "on-sort"          => "dataCtrl.onSort(headerId, isAscending)",
+      "load-more-items"  => "dataCtrl.onLoadNext(start, perPage)",
+      "on-item-selected" => "dataCtrl.onItemSelect(item, isSelected)",
+      "columns"          => "dataCtrl.gtlData.cols"
+    )
+  end
+
+  # This method collects all the data comming from side channels
+  # so that `render_gtl` can be a pure function.
+  #
+  def render_gtl_outer(no_flash_div)
+    parent_id = if @report_data_additional_options
+                  @report_data_additional_options[:parent_id]
```

Comment:
consider using try ?

Reply:
yes, that makes sense, let's do that in the first refactoring round of the helper.